### PR TITLE
Add undo feature

### LINKED ISSUE #146: As a home-based beauty salon owner, I want to undo my previous command

### PR DESCRIPTION
**Pull request by hanstirtaputra**
Undo feature reverts the state of the model to the last **changed** state. Undo command does not undo commands that does not change the state of the `ModelManager`, for example `listsvc` or `findsvc`.

### CODE REVIEW

File: src/main/java/seedu/homerce/logic/commands/UndoCommand.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import seedu.homerce.model.Model;
+
+/**
+ * Undo the last command that updates the state of Homerce's storage.
+ *
+ * A command that does not affect the state of Homerce's storage will not be considered in the undo command.
+ */
+public class UndoCommand extends Command {
+    public static final String COMMAND_WORD = "undo";
+    public static final String MESSAGE_SUCCESS = "You have undone you previous change!";
```
Comment (RuiFengg) [inline src/main/java/seedu/homerce/logic/commands/UndoCommand.java]: Message takes a bit of time to understand for me, maybe undo success or undo complete is sufficient 😸

File: src/main/java/seedu/homerce/logic/commands/UndoCommand.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    @Override
+    public CommandResult execute(Model model, HistoryManager historyManager) throws CommandException {
+        Model previousModel = historyManager.getPreviousState();
+        if (previousModel == null) {
+            throw new CommandException("There are no more changes to be undone!");
```
Comment (RuiFengg) [inline src/main/java/seedu/homerce/logic/commands/UndoCommand.java]: Same here, maybe "Already at oldest change" (copied from vscode vim)

**Review by RuiFengg — APPROVED** [review]
LGTM, but can consider changing the message to user 😸

File: src/main/java/seedu/homerce/logic/commands/UndoCommand.java (hunk 2 above)
Comment (hanstirtaputra) [reply src/main/java/seedu/homerce/logic/commands/UndoCommand.java]: Okay, thanks for this, I've updated the response messages accordingly!

**Pull request merged**